### PR TITLE
Random code cleanup

### PR DESCRIPTION
**Pull request by janmulder**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ ] New feature
- [x] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:
This a set of small cleanup commits, mostly for code that is not used at all, so that can silently be removed. Also added 2 commits to silence compiler warnings. The careful reader might notice that most commits here are from May 2017, and yes, it got lost in space, and just found this branch back, rebased it, cleaned it up to the latest insights.

### Additional information:
Well, it compiles, so it is correct :-)

### CODE REVIEW

File: core/liquivision.c [hunk 1]
```diff
@@ -310,6 +310,7 @@ static void parse_dives (int log_version, const unsigned char *buf, unsigned int
 
 		unsigned int event_code, d = 0, e;
 		struct lv_event event;
+		memset(&event, 0, sizeof(event));
```
Comment (bstoeger) [inline core/liquivision.c:313]: This does not hurt - but the warning is bogus.

The code accessing `event` will only be reached if `event_code == 0xf` in which case `handle_event_ver3()` will have filled out the appropriate fields. Usually gcc is quite smart about these things. I wonder why it is missing it in this case.

File: core/uemis.c [hunk 2]
```diff
@@ -374,8 +373,8 @@ void uemis_parse_divelog_binary(char *base64, void *datap)
 	/* get data from the footer */
 	char buffer[24];
 
-	snprintf(version, sizeof(version), "%1u.%02u", data[18], data[17]);
-	add_extra_data(dc, "FW Version", version);
+	snprintf(buffer, sizeof(buffer), "%1u.%02u", data[18], data[17]);
```
Comment (bstoeger) [inline core/uemis.c:376]: The format string `%1u` is strange - isn't this equivalent to `%u`?

Likewise, two occurrences of `%0u` later in the same file could be written as `%u`, if I'm not mistaken.

**Comment by dirkhh** [comment]
No. %1u and %02u mean that you want one unsigned digit and then two unsigned digits, 0-padded

**Comment by bstoeger** [comment]
AFAIK `%1u` means "at least one digit", which is tautological since I don't see how `%u` would ever produce less than one digit. A quick test seems to confirm this.

If this is correct, `%0u` would mean something like: Don't pad, but if you would pad, use zero instead of space.

`%02u`, on the other hand, is fine: Print at least two digits, but zero-pad in case there is only one.

No?

**Comment by bstoeger** [comment]
In any case, it shouldn't make a difference, so I don't see why this PR shouldn't be applied as-is.

**Pull request merged**